kvdb/src/mock_param_dict_impl.cpp

```diff
@@ -0,0 +1,150 @@
+
+std::vector<float> ParamDict::GetSparseValue(int64_t feasign, int64_t slot) {
+    return this->GetSparseValue(std::to_string(feasign), std::to_string(slot));
+}
+
+bool ParamDict::InsertSparseValue(int64_t feasign, int64_t slot, const std::vector<float>& values) {
+    return this->InsertSparseValue(std::to_string(feasign), std::to_string(slot), values);       
+}
+
+bool ParamDict::InsertSparseValue(std::string feasign, std::string slot, const std::vector<float>& values) {
+    auto FloatToBytes = [](float fvalue, uint8_t *arr){
+        unsigned char  *pf = nullptr;
+        unsigned char *px = nullptr;
+        unsigned char i = 0;
+        pf =(unsigned char *)&fvalue;
+        px = arr;
+        for (i = 0; i < 4; i++)
+        {
+            *(px+i)=*(pf+i);
+        }
+    };
+
+    std::string key = feasign + slot;
+    uint8_t* values_ptr = new uint8_t[values.size() * 4];
+    std::string value;
+    for (size_t i = 0; i < values.size(); i++) {
+        FloatToBytes(values[i], values_ptr + 4 * i);
+    }
+    char* raw_values_ptr = reinterpret_cast<char*>(values_ptr);
+    for (size_t i = 0; i < values.size()*4; i++) {
+        value.push_back(raw_values_ptr[i]);
+    }
+    back_db->Set(key, value);
+//TODO: change stateless to stateful
+    return true;
+}
+
+void ParamDict::UpdateBaseModel() {
+    auto is_number = [] (const std::string& s)
+    {
+        return !s.empty() && std::find_if(s.begin(), 
+                        s.end(), [](char c) { return !std::isdigit(c); }) == s.end();
+    };
+   std::thread t([&] () {
+        for (FileReaderPtr file_reader: this->file_reader_lst_) {
+            std::string line;
+            std::ifstream infile(file_reader->GetFileName());
+            if (infile.is_open()) {
+                while (getline(infile, line)) {
+                    std::pair<Key, Value> kvpair = read_func_(line);
+                    std::vector<float> nums;
+                    for (size_t i = 0; i < kvpair.second.size(); i++) {
+                        if (is_number(kvpair.second[i])) {
+                            nums.push_back(std::stof(kvpair.second[i]));
+                        }
+                    }
+                    this->InsertSparseValue(kvpair.first, "", nums);
+                }
+            }
+            infile.close();
+        }
+        AbsKVDBPtr temp = front_db;
+        front_db = back_db;
+        back_db = temp;
+   });
+   t.detach();
+}
+
+
+void ParamDict::UpdateDeltaModel() {
+    UpdateBaseModel();
+}
+
+std::pair<AbsKVDBPtr, AbsKVDBPtr> ParamDict::GetKVDB()  {
+    return {front_db, back_db};
+}
+
+void ParamDict::SetKVDB(std::pair<AbsKVDBPtr, AbsKVDBPtr> kvdbs) {
+    this->front_db = kvdbs.first;
+    this->back_db = kvdbs.second;
+}
+
+void ParamDict::CreateKVDB() {
+    this->front_db = std::make_shared<RocksKVDB>();
+    this->back_db = std::make_shared<RocksKVDB>();
+    this->front_db->CreateDB();
+    this->back_db->CreateDB();
+}
+
+ParamDict::~ParamDict() {
+
+}
+
+
```

Comment:
Redundant blank lines

Reply:
redundant line issue has been auto-fix by pre-commit. thank you.